backend/app/api/v1/commons/ocm.py

```diff
@@ -3,35 +3,40 @@
 from app.services.search import ElasticService
 
 
-async def getData(start_datetime: date, end_datetime: date, configpath: str):
+async def getData(
+    start_datetime: date, end_datetime: date, size: int, offset: int, configpath: str
+):
     query = {
+        "size": size,
+        "from": offset,
         "query": {
             "bool": {
                 "filter": {"range": {"metrics.earliest": {"format": "yyyy-MM-dd"}}}
             }
-        }
+        },
     }
 
     es = ElasticService(configpath=configpath)
     response = await es.post(
         query=query,
+        size=size,
         start_date=start_datetime,
         end_date=end_datetime,
         timestamp_field="metrics.earliest",
     )
     await es.close()
-    tasks = [item["_source"] for item in response]
+    tasks = [item["_source"] for item in response["data"]]
     jobs = pd.json_normalize(tasks)
     if len(jobs) == 0:
-        return jobs
+        return {"data": jobs, "total": response["total"]}
 
     if "buildUrl" not in jobs.columns:
         jobs.insert(len(jobs.columns), "buildUrl", "")
     if "ciSystem" not in jobs.columns:
         jobs.insert(len(jobs.columns), "ciSystem", "")
     jobs.fillna("", inplace=True)
     jobs["jobStatus"] = jobs.apply(convertJobStatus, axis=1)
```

Comment:
I don't find long conditional blocks necessarily easier to follow. In the former case there was the one-line early return vs 2 indented conditional lines for a common return, and I think that's a good call. This block is getting big enough that the early return might be more readable... although there are probably much better ways to refactor the method to avoid this sort of thing in the future.